File: tests/examples_tests/python/wrappers/containers/containers_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "examples/cxx/containers/containers.hpp"

namespace py = pybind11;

void bindItem(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::Item> item(m, "Item");

    item.def_readwrite("value", &iegen::example::Item::value);

    item.def(py::init<int>(), py::arg("_value"));
}
void bindContainers(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::Containers> containers(m, "Containers");

    containers.def(py::init<>());
    containers.def("add_string_pair", &iegen::example::Containers::addStringPair, py::arg("info"));
    containers.def("add_int_map", &iegen::example::Containers::addIntMap, py::arg("info"));
    containers.def("add_string_map", &iegen::example::Containers::addStringMap, py::arg("info"));
    containers.def("add_int_pair", &iegen::example::Containers::addIntPair, py::arg("info"));
    containers.def("get_string_pair", &iegen::example::Containers::getStringPair);
    containers.def("get_string_map", &iegen::example::Containers::getStringMap);
    containers.def("get_int_map", &iegen::example::Containers::getIntMap);
    containers.def("get_map", &iegen::example::Containers::getMap);
    containers.def("add_int_vector", &iegen::example::Containers::addIntVector, py::arg("v"));
    containers.def("add_string_vector", &iegen::example::Containers::addStringVector, py::arg("v"));
    containers.def("get_string_vector", &iegen::example::Containers::getStringVector);
    containers.def("get_int_vector", &iegen::example::Containers::getIntVector);
}

File: examples/python/containers/containers_pygen.py
from __future__ import annotations
from typing import *
import iegen.examples.containers.containers as pybind_containers_pygen
from bind_utils import *
from metaclass import *


class Item(metaclass=OriginalMethodsMetaclass):
    """
    comments
    """

    @bind
    def __init__(self, _value: int):
        pass

    @property
    @bind
    def value(self) -> int:
        """
        comments
        """
        pass

    @value.setter
    @bind
    def value(self, value: int):
        """
        comments
        """
        pass


class Containers(metaclass=OriginalMethodsMetaclass):
    """
    comments
    """

    @bind
    def __init__(self):
        pass

    @bind
    def add_string_pair(self, info: tuple) -> None:
        """
        comments
        """
        pass

    @bind
    def add_int_map(self, info: dict) -> None:
        """
        comments
        """
        pass

    @bind
    def add_string_map(self, info: dict) -> None:
        """
        comments
        """
        pass

    @bind
    def add_int_pair(self, info: tuple) -> None:
        """
        comments
        """
        pass

    @bind
    def get_string_pair(self) -> tuple:
        """
        comments
        """
        pass

    @bind
    def get_string_map(self) -> dict:
        """
        comments
        """
        pass

    @bind
    def get_int_map(self) -> dict:
        """
        comments
        """
        pass

    @bind
    def get_map(self) -> dict:
        """
        comments
        """
        pass

    @bind
    def add_int_vector(self, v: list) -> None:
        """
        comments
        """
        pass

    @bind
    def add_string_vector(self, v: list) -> None:
        """
        comments
        """
        pass

    @bind
    def get_string_vector(self) -> list:
        """
        comments
        """
        pass

    @bind
    def get_int_vector(self) -> list:
        """
        comments
        """
        pass
